parsers/gdscript.c

```diff
@@ -0,0 +1,1275 @@
+		return false;
+
+	name = newToken ();
+	copyToken (name, token);
+
+	readToken (token);
+	/* collect parameters or inheritance */
+	if (token->type == '(')
+	{
+		int prevTokenType = token->type;
+		int depth = 1;
+
+		arglist = vStringNew ();
+		if (kind != K_CLASS)
+			reprCat (arglist, token);
+
+		do
+		{
+			if (token->type != TOKEN_WHITESPACE &&
+			    token->type != '*')
+			{
+				prevTokenType = token->type;
+			}
+
+			readTokenFull (token, true);
+			if (kind != K_CLASS || token->type != ')' || depth > 1)
+				reprCat (arglist, token);
+
+			if (token->type == '(' ||
+			    token->type == '[' ||
+			    token->type == '{')
+				depth ++;
+			else if (token->type == ')' ||
+			         token->type == ']' ||
+			         token->type == '}')
+				depth --;
+			else if (kind != K_CLASS && depth == 1 &&
+			         token->type == TOKEN_IDENTIFIER &&
+			         (prevTokenType == '(' || prevTokenType == ',') &&
+			         parameterCount < ARRAY_SIZE (parameterTokens) &&
+			         GDScriptKinds[K_PARAMETER].enabled)
+			{
+				tokenInfo *parameterName = newToken ();
+
+				copyToken (parameterName, token);
+				parameterTokens[parameterCount] = parameterName;
+				parameterTypes [parameterCount++] = parseParamTypeAnnotation (token, arglist);
+			}
+		}
+		while (token->type != TOKEN_EOF && depth > 0);
+	}
+
+	if (kind == K_CLASS)
+		corkIndex = makeClassTag (name, arglist);
+	else
+		corkIndex = makeFunctionTag (name, arglist, decorators);
+
+	lv = nestingLevelsPush (GDScriptNestingLevels, corkIndex);
+	PY_NL (lv)->indentation = token->indent;
+
+	deleteToken (name);
+	vStringDelete (arglist);
+
+	if (parameterCount > 0)
+	{
+		unsigned int i;
+
+		for (i = 0; i < parameterCount; i++)
+		{
+			int paramCorkIndex = makeSimpleGDScriptTag (parameterTokens[i], K_PARAMETER);
+			deleteToken (parameterTokens[i]);
+			tagEntryInfo *e = getEntryInCorkQueue (paramCorkIndex);
+			if (e && parameterTypes[i])
+			{
+				e->extensionFields.typeRef [0] = eStrdup ("typename");
+				e->extensionFields.typeRef [1] = vStringDeleteUnwrap (parameterTypes[i]);
+				parameterTypes[i] = NULL;
+			}
+			vStringDelete (parameterTypes[i]); /* NULL is acceptable. */
+		}
+	}
+
+	tagEntryInfo *e;
+	vString *t;
+	if (kind != K_CLASS
+		&& (e = getEntryInCorkQueue (corkIndex))
+		&& (t = parseReturnTypeAnnotation (token)))
+	{
+		e->extensionFields.typeRef [0] = eStrdup ("typename");
+		e->extensionFields.typeRef [1] = vStringDeleteUnwrap (t);
+	}
+
+	return true;
+}
+
+static bool parseEnum (tokenInfo *const token)
+{
+	vString *arglist = NULL;
+	tokenInfo *name = newToken();
+	tokenInfo *parameterTokens[16] = { NULL };
+	unsigned int parameterCount = 0;
+	int corkIndex;
+
+	readToken (token);
+
+	if (token->type == '{') {
+		copyToken (name, token);
+		vString *enumName = anonGenerateNew ("anon_enum_", K_ENUM);
+		vStringClear(name->string);
+		vStringCatS(name->string, vStringValue(enumName));
+		vStringDelete(enumName);
+		name->type = TOKEN_IDENTIFIER;
```

Comment:
You don't have to make a new vstring.
Instead, let's reuse name->string area.
```
diff --git a/parsers/gdscript.c b/parsers/gdscript.c
index f9f83f206..e912ea28f 100644
--- a/parsers/gdscript.c
+++ b/parsers/gdscript.c
@@ -918,10 +918,8 @@ static bool parseEnum (tokenInfo *const token)
 
        if (token->type == '{') {
                copyToken (name, token);
-               vString *enumName = anonGenerateNew ("anon_enum_", K_ENUM);
-               vStringClear(name->string);
-               vStringCatS(name->string, vStringValue(enumName));
-               vStringDelete(enumName);
+               vStringClear (name->string);
+               anonGenerate (name->string, "anon_enum_", K_ENUM);
                name->type = TOKEN_IDENTIFIER;
        } else if (token->type == TOKEN_IDENTIFIER) {
                copyToken (name, token);
```

Reply:
Oh, anonGenerate doing that makes it a good bit nicer